### ISSUE TITLE
Att&ck matrix appears broken after recent update

### ISSUE THREAD

**Issue by DigiAngel** [MISP/MISP#4393]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Support
| OS version (server)       | Ubuntu 18.04
| OS version (client)       | Ubuntu 18.04
| PHP version               | php7.2
| MISP version / git hash   | 0e4f66ee5722590d341ed2825101a11dd80e02d9
| Browser                   | Firefox

### Expected behavior
See and manipulate the Att&ck matrix

### Actual behavior
Black bar

### Steps to reproduce the behavior
Click Att&ck matrix

### Logs, screenshots, configuration dump, ...
![Peek 2019-03-28 06-12](https://user-images.githubusercontent.com/947191/55156880-a6c14c80-5120-11e9-809e-59976e941487.gif)

~~~
2019-03-28 06:07:22 Error: [Exception] Galaxy cannot be represented as a matrix
Request URL: /events/viewGalaxyMatrix/854/0/event/1
Stack Trace:
#0 /var/www/MISP/app/Controller/EventsController.php(4849): Galaxy->getMatrix('0')
#1 [internal function]: EventsController->viewGalaxyMatrix('854', '0', 'event', '1')
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(499): ReflectionMethod->invokeArgs(Object(EventsController), Array)
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction(Object(CakeRequest))
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke(Object(EventsController), Object(CakeRequest))
#5 /var/www/MISP/app/webroot/index.php(92): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#6 {main}
2019-03-28 06:07:23 Error: [Exception] Galaxy cannot be represented as a matrix
Request URL: /events/viewGalaxyMatrix/854/0/event/1
Stack Trace:
#0 /var/www/MISP/app/Controller/EventsController.php(4849): Galaxy->getMatrix('0')
#1 [internal function]: EventsController->viewGalaxyMatrix('854', '0', 'event', '1')
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(499): ReflectionMethod->invokeArgs(Object(EventsController), Array)
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction(Object(CakeRequest))
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke(Object(EventsController), Object(CakeRequest))
#5 /var/www/MISP/app/webroot/index.php(92): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#6 {main}
~~~

**Comment by mokaddem** [comment]
Hello,
Fetching the latest version of ``misp-galaxy`` and updating should solve the issue.
Please, re-open if it doesn't.
Cheers!

**Comment by DigiAngel** [comment]
So....my process for updating has been:
~~~
cd /var/www/MISP
git pull origin 2.4
git submodule update --init --force
chown -R www-data:www-data /var/www/MISP/
~~~
Iv'e never git pulled/cloned misp-galaxy.  I followed:
[install](https://github.com/MISP/MISP/blob/2.4/docs/INSTALL.ubuntu1804.md)

how do I up install/update misp-galaxy?  Thank you.

**Comment by DigiAngel** [comment]
Also of interest, there doesn't appear to be a way to reopen this ticket :(

**Comment by iglocska** [comment]
That looks correct for updating the submodule, however, you still need to ingest it within misp. Just go to your galaxy index and click on force update galaxies.

**Comment by DigiAngel** [comment]
Thank you @iglocska that did the trick!

**Comment by jstnk9** [comment]
> That looks correct for updating the submodule, however, you still need to ingest it within misp. Just go to your galaxy index and click on force update galaxies.

I have the same problem and the solution doesn't work for me.

My MISP version is v2.4.117.

Thanks!!

**Comment by mokaddem** [comment]
Hey @jstnk9,
- What is the output of ``servers/serverSettings/diagnostics`` under ``Submodule version``?
- Also what are the meta of the Mitre ``Attack Pattern galaxy``? (under ``galaxies/view/X``)

**Comment by jstnk9** [comment]
> Hey @jstnk9,
> 
> * What is the output of `servers/serverSettings/diagnostics` under `Submodule version`?
> * Also what are the meta of the Mitre `Attack Pattern galaxy`? (under `galaxies/view/X`)

Hey @mokaddem !

Under submodule versión i have the nex image:
![image](https://user-images.githubusercontent.com/7794663/68202326-c9bfb300-ffc3-11e9-9ace-14e915ad780e.png)

And i have the next galaxies:
![image](https://user-images.githubusercontent.com/7794663/68202653-65e9ba00-ffc4-11e9-835c-abc8f7e827dc.png)

This is the request to the server:
![image](https://user-images.githubusercontent.com/7794663/68202676-70a44f00-ffc4-11e9-87bc-6d149090d693.png)

**Comment by mokaddem** [comment]
Thanks for the output.
I'm still missing this one:
- what are the meta of the Mitre Attack Pattern galaxy (so under galaxies/view/89 I guess)?

**Comment by jstnk9** [comment]
> Thanks for the output.
> I'm still missing this one:
> 
> * what are the meta of the Mitre Attack Pattern galaxy (so under galaxies/view/89 I guess)?

Hi @mokaddem 

here is the information:

![image](https://user-images.githubusercontent.com/7794663/68210176-4444fe80-ffd5-11e9-9f95-075cfeac623c.png)

![image](https://user-images.githubusercontent.com/7794663/68210187-4eff9380-ffd5-11e9-9648-108b9339850f.png)


Thanks a lot.

**Comment by mokaddem** [comment]
Hey @jstnk9,
We did not forget you. ;)
The past weeks have been a bit crazy due to the various end-of-the-year events.
I drafted a diagnostic tool for the galaxies (and JSON related libraries) that will allow us to find out what your issue is.
Hopefully, it will be available soon...
Thanks for your patience!

**Comment by jstnk9** [comment]
> Hey @jstnk9,
> We did not forget you. ;)
> The past weeks have been a bit crazy due to the various end-of-the-year events.
> I drafted a diagnostic tool for the galaxies (and JSON related libraries) that will allow us to find out what your issue is.
> Hopefully, it will be available soon...
> Thanks for your patience!

Thanks so much mokaddem :)

**Comment by jstnk9** [comment]
> Hey @jstnk9,
> We did not forget you. ;)
> The past weeks have been a bit crazy due to the various end-of-the-year events.
> I drafted a diagnostic tool for the galaxies (and JSON related libraries) that will allow us to find out what your issue is.
> Hopefully, it will be available soon...
> Thanks for your patience!

Hi @mokaddem !

news?

Thanks! :)

**Comment by mokaddem** [comment]
Thanks for the nudge,
Unfortunately, the tool is not ready to be merged into 2.4 yet. But we can still give it a try if these steps doesn't work, as multiple things might have changed in the meantime:
1. Could you perform an usual update (make sure the perm are correct afterward)
```
git pull origin 2.4
git submodule update --init --force
```
2. Could you paste the output of the diagnostic page (the part with the submodules) to check the commit ids?

Thanks a lot!
